### PR TITLE
Provide suggestions for common cannot-find-name errors

### LINKED ISSUE #25309: Better hints for 'cannot find name' errors
Discussed at #25306

We found that for "Cannot find name" errors, certain identifiers were fairly frequent:

> `require`, `map`, `buffer`, `describe`, `$`, `asyncIterator`, `process`, `symbol`, `module`, `promise`

We're thinking that we can elaborate beyond the usual error message.

* `$` => Suggest jQuery
* `describe` => Did you mean to have types for mocha, jest, etc.?
* `process`, `require`, `module` => Suggest including Node types
* `Symbol`, `Promise`, `Map`, `Set`, `WeakMap`, `WeakSet`, `Iterator`, `AsyncIterator` => User needs to set a different `lib` or `target`.

### PR DESCRIPTION
**Pull request by weswigham**
Fixes #25309

### CODE REVIEW

**Comment by weswigham** [comment]
cc @DanielRosenwasser

File: tests/baselines/reference/didYouMeanSuggestionErrors.errors.txt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+tests/cases/compiler/didYouMeanSuggestionErrors.ts(17,23): error TS2583: Cannot find name 'Set'. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
+tests/cases/compiler/didYouMeanSuggestionErrors.ts(18,23): error TS2583: Cannot find name 'WeakMap'. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
+tests/cases/compiler/didYouMeanSuggestionErrors.ts(19,23): error TS2583: Cannot find name 'WeakSet'. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
+tests/cases/compiler/didYouMeanSuggestionErrors.ts(20,19): error TS2585: 'Symbol' only refers to a type, but is being used as a value here. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
+tests/cases/compiler/didYouMeanSuggestionErrors.ts(21,19): error TS2585: 'Promise' only refers to a type, but is being used as a value here. Do you need to change your target library? Try changing the `lib` compiler option to es2015 or later.
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/didYouMeanSuggestionErrors.errors.txt]: ❤️

**Comment by DanielRosenwasser** [comment]
Can't wait for the quick fixes on this.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -13818,6 +13821,36 @@ namespace ts {
...
+                    return Diagnostics.Cannot_find_name_0_Do_you_need_to_install_type_definitions_for_node_Try_npm_i_types_Slashnode;
+                case "Map":
+                case "Set":
+                case "Promise":
+                case "Symbol":
+                case "WeakMap":
+                case "WeakSet":
+                case "Iterator":
+                case "AsyncIterator":
+                    return Diagnostics.Cannot_find_name_0_Do_you_need_to_change_your_target_library_Try_changing_the_lib_compiler_option_to_es2015_or_later;
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: `EventEmitter`?

File: src/compiler/checker.ts [hunk 3]
```diff
@@ -13818,6 +13821,36 @@ namespace ts {
...
+                    return Diagnostics.Cannot_find_name_0_Do_you_need_to_install_type_definitions_for_jQuery_Try_npm_i_types_Slashjquery;
+                case "describe":
+                case "suite":
+                case "it":
+                case "test":
+                    return Diagnostics.Cannot_find_name_0_Do_you_need_to_install_type_definitions_for_a_test_runner_Try_npm_i_types_Slashjest_or_npm_i_types_Slashmocha;
+                case "process":
+                case "require":
+                case "Buffer":
+                case "module":
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: `EventEmitter`?

Comment (weswigham) [reply src/compiler/checker.ts]: mmmmm it's not actually a global afaik, but i'll add it if you think it's relevant

Comment (DanielRosenwasser) [reply src/compiler/checker.ts]: Nah this is fine. 🚢 🇮🇹

**Pull request merged**